app/src/androidTest/java/com/android/sample/ui/activityDetails/CommentItemTest.kt

```diff
@@ -147,41 +148,58 @@ class CommentItemTest {
   }
 
   @Test
-  fun replyIsPosted_whenReplyInputIsFilledAndPosted() {
-    var replyPosted = false
+  fun repliesAreDisplayedCorrectly() {
     composeTestRule.setContent {
       CommentItem(
           profileId = testProfileId,
           comment = testComment,
-          onReplyComment = { replyText, _ -> replyPosted = replyText.isNotBlank() },
+          onReplyComment = { _, _ -> },
           onDeleteComment = {})
     }
 
-    composeTestRule
-        .onNodeWithTag("ReplyButton_${testComment.uid}")
-        .performClick() // Use the test tag
-    composeTestRule
-        .onNodeWithTag("replyInputField_${testComment.uid}")
-        .assertIsDisplayed() // Also ensure to set a test tag for the input field
-    composeTestRule
-        .onNodeWithTag("replyInputField_${testComment.uid}")
-        .performTextInput("New reply")
-    composeTestRule
-        .onNodeWithTag("postReplyButton_${testComment.uid}")
-        .performClick() // Use the test tag
-    assert(replyPosted)
+    composeTestRule.onNodeWithText("John: This is a reply").assertIsDisplayed()
   }
 
   @Test
-  fun repliesAreDisplayedCorrectly() {
+  fun replyButton_isNotDisplayedForReplies() {
+    val replyComment = testComment.replies.first() // The reply to the original comment
     composeTestRule.setContent {
       CommentItem(
           profileId = testProfileId,
-          comment = testComment,
+          comment = replyComment, // Pass in the reply comment
           onReplyComment = { _, _ -> },
-          onDeleteComment = {})
+          onDeleteComment = {},
+          allowReplies = false // Disable replies for this test
+          )
     }
 
+    // Ensure that the reply button is not displayed for replies
+    composeTestRule.onNodeWithTag("ReplyButton_${replyComment.uid}").assertDoesNotExist()
+  }
```

Comment:
Great job implementing the test for the new feature that disables the reply button.